libs/server/controllers/leaderboard/src/lib/leaderboard.service.ts

```diff
@@ -0,0 +1,25 @@
+import { PrismaService } from '@bregenz-bewegt/server-prisma';
+import {
+  Competitor,
+  LeaderboardPaginationQueryDto,
+} from '@bregenz-bewegt/shared/types';
+import { Injectable } from '@nestjs/common';
+import { Role } from '@prisma/client';
+
+@Injectable()
+export class LeaderboardService {
+  constructor(private prismaService: PrismaService) {}
+
+  async getLeaderboard({
+    skip,
+    take,
+  }: LeaderboardPaginationQueryDto): Promise<Competitor[]> {
+    return this.prismaService.user.findMany({
+      skip,
+      take,
+      where: { role: { not: Role.GUEST } },
+      select: { username: true, coins: true },
+      orderBy: { coins: 'desc' },
```

Comment:
es fehlt no die implementierung der filterung nach jahren

Reply:
stimmt. solli des do glei dazua maha? i hobs bis jetzt usglo weil i ned gwisst hob wie genau ma jetzt belohnt wird und so. wenn zb in quartalen abgrechnet wird wärs eventuall guat o in deana intervallen die filteroptionen zum mappen.

Reply:
jo, i würds gleich dazua macha - in jahren

Reply:
passt denn mahi des no

Reply:
mir isch gad ufgfalla für die implementierung für da jahresfilter müsst ma zersch no des coin system in da datenbank ändra. mir hon no kua activities die ind datenbank gschriba wörran. zum jetziga stand kanni des leider no ned maha.